.github/workflows/compat-docker-release.yml

```diff
@@ -12,7 +14,7 @@ jobs:
     name: Docker release - Linux::x86_64
     uses: ./.github/workflows/docker_linux_x86_64.yml
     with:
-      lightning-version: v0.37.0
-      pennylane-version: v0.37.0-rc0
+      lightning-version: v0.38.0_rc
```

Comment:
@vincentmr What's the reason behind using underscore instead of hyphen? I reckon it's not consistent across PL rc branches.

Reply:
It is our convention.

Reply:
really! :grimacing: